From 71aaec92de26da29a26d58e734bcbd0ad5ea122f Mon Sep 17 00:00:00 2001
From: Ling556 <ling_haner@163.com>
Date: Thu, 23 May 2024 23:43:23 +0800
Subject: [PATCH] kick all player on stop


diff --git a/src/main/java/net/minecraft/server/LoginListener.java b/src/main/java/net/minecraft/server/LoginListener.java
index be43d9292..2d83c0879 100644
--- a/src/main/java/net/minecraft/server/LoginListener.java
+++ b/src/main/java/net/minecraft/server/LoginListener.java
@@ -56,6 +56,12 @@ public class LoginListener implements PacketLoginInListener, ITickable {
     }
 
     public void e() {
+        // Custom start - Do not allow logins while the server is pre shutting down
+        if(MinecraftServer.getServer().isPreShutdown()){
+            this.disconnect(new ChatMessage(org.spigotmc.SpigotConfig.restartMessage));
+            return;
+        }
+        // Custom end
         // Paper start - Do not allow logins while the server is shutting down
         if (!MinecraftServer.getServer().isRunning()) {
             this.disconnect(new ChatMessage(org.spigotmc.SpigotConfig.restartMessage));
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 3b982f990..4d4aa065c 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -49,6 +49,7 @@ import org.bukkit.craftbukkit.Main;
 // CraftBukkit end
 import org.spigotmc.SlackActivityAccountant; // Spigot
 import co.aikar.timings.MinecraftTimings; // Paper
+import org.spigotmc.SpigotConfig;
 
 public abstract class MinecraftServer implements ICommandListener, Runnable, IAsyncTaskHandler, IMojangStatistics {
 
@@ -69,6 +70,7 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
     private int u = -1;
     public WorldServer[] worldServer;
     private PlayerList v;
+    private boolean isPreShutdown = false; // Custom - flag to signify shutdown kick player
     private boolean isRunning = true;
     private boolean isRestarting = false; // Paper - flag to signify we're attempting to restart
     private boolean isStopped;
@@ -550,12 +552,41 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
         return this.isRunning;
     }
 
+    // Custom start
+    public boolean isPreShutdown(){
+        return this.isPreShutdown;
+    }
+    // Custom end
+
     // Paper start - allow passing of the intent to restart
     public void safeShutdown() {
         safeShutdown(false);
     }
 
     public void safeShutdown(boolean isRestarting) {
+        // Custom start
+        // Kick all players
+        MinecraftServer.LOGGER.info("Kicking all players");
+        this.isPreShutdown = true;
+        for ( EntityPlayer p : com.google.common.collect.ImmutableList.copyOf( MinecraftServer.getServer().getPlayerList().players ) )
+        {
+            p.playerConnection.disconnect(SpigotConfig.restartMessage);
+        }
+        // Give the socket a chance to send the packets
+        try
+        {
+            Thread.sleep( 100 );
+        } catch ( InterruptedException ex ) {}
+
+        this.getServerConnection().b();
+
+        // Give the socket a chance to send the packets
+        try
+        {
+            Thread.sleep( 10000 );
+        } catch ( InterruptedException ex ) {}
+
+        // Custom end
         this.isRunning = false;
         this.isRestarting = isRestarting;
     }
-- 
2.34.1

