From a2b38a73e4a6f14ad7d135ecd175df147f033d37 Mon Sep 17 00:00:00 2001
From: Ling556 <ling_haner@163.com>
Date: Thu, 23 May 2024 23:48:41 +0800
Subject: [PATCH] Remove inventory drag action


diff --git a/src/main/java/net/minecraft/server/Container.java b/src/main/java/net/minecraft/server/Container.java
index 3a488ce20..8c305c49d 100644
--- a/src/main/java/net/minecraft/server/Container.java
+++ b/src/main/java/net/minecraft/server/Container.java
@@ -148,6 +148,7 @@ public abstract class Container {
                     this.h.add(slot);
                 }
             } else if (this.g == 2) {
+                /* Custon start
                 if (!this.h.isEmpty()) {
                     itemstack2 = playerinventory.getCarried().cloneItemStack();
                     l = playerinventory.getCarried().getCount();
@@ -211,7 +212,13 @@ public abstract class Container {
                         ((EntityPlayer) entityhuman).updateInventory(this);
                     }
                     // CraftBukkit end
+                }*/
+                // Custom end
+                // Custom start
+                if (entityhuman instanceof EntityPlayer) {
+                    ((EntityPlayer) entityhuman).updateInventory(this);
                 }
+                // Custom end
 
                 this.d();
             } else {
-- 
2.34.1

